framework_tvb/tvb/interfaces/web/controllers/simulator_controller.py

```diff
@@ -697,6 +697,14 @@ def _skip_raw_monitor(self, monitors):
                                                           type(monitors[first_monitor_index]).__name__)
         return first_monitor_index, last_loaded_fragment_url
 
+    @staticmethod
+    def _check_cortical_for_spatial_average(connectivity, form, is_surface_simulation):
+        connectivity_index = ABCAdapter.load_entity_by_gid(connectivity)
+        connectivity = h5.load_from_index(connectivity_index)
```

Comment:
This load is inefficient. 
Can't we prepare a pre-computed flag on ConnectivityIndex and evaluate it when we import a Connectivity (in `ConnectivityIndex.fill_from_has_traits`? 
The situation won't ever change in TVB after import, so no need to evaluate at every sim execution time

Reply:
Okay, I implemented this change, but that adds a new field to the database for ConnectivityIndex. Is it okay?

Reply:
yes, it is ok to have extra extra fields in *Index entities, to accommodate our use-cases. why wouldn't be ?

Reply:
Solved.